messageService/message.go

```diff
@@ -0,0 +1,90 @@
+package messageService
+
+import (
+	"time"
+	"encoding/json"
+	"fmt"
+	"log"
+)
+
+//server address, IP and port will fill later
+const server = "IP:port"
+
+//structure for message(should be in JSON format) which consist with header and body
+type Message struct {
+	Header MessageHeader `json:"header"`
+	Body MessageBody `json:"body"`
+}
+
+// header for Message
+type MessageHeader struct {
+	Type_ string `json:"type"`// this named "Type_" because "type" - keyword
+	Command string `json:"command"` //depend on action from user (e.g. sent message, set info e.t.c.)
+	Sender int `json:"sender"`
+	Receiver int `json:"receiver"`
+	Time time.Time `json:"time"`
+	Auth string `json:"auth"`// value of Auth is "token" at the beginning (maybe use JSON web token?)
+}
+
+// body for message
+type MessageBody struct {
+	Text string `json:"text"`
+	//links []link -> slice of formatted URL's
+}
+
+//func unmarshal message ([] byte JSON -> Message)
+func UnmarshalMessage(byteMessage [] byte ) Message {
+	var message Message
+	err := json.Unmarshal(byteMessage, &message)
+	if err != nil{
+		log.Println("Can't unmarshal this array of bytes :", err)
```

Comment:
Error should be returned to caller, otherwise he would not be able to cancel execution if message is corrupted.